src/main/java/uk/gov/ons/ctp/response/casesvc/service/impl/CaseServiceImpl.java

```diff
@@ -219,6 +219,13 @@ else if (caseEvent.getCategory().equals(CategoryDTO.CategoryName.DISABLE_RESPOND
         }
       }
 
+      else if (caseEvent.getCategory().equals(CategoryDTO.CategoryName.REPLACED) && category.getNewCaseSampleUnitType().equals(SampleUnitType.B))  {
```

Comment:
A good refactoring option here would be changing this entire if/else if/else if block to a `switch(caseEvent.getCategory())`. This particular line is the only case that would cause a problem in doing this refactoring and can be simply changed as follows:

`case REPLACED:
if category.getNewCaseSampleUnitType().equals(SampleUnitType.B)) {
 // do stuff
}`